ls1intum/Artemis
# `Exam mode`: Fix the number of complaints in the exam assessment dashboard

### Checklist
#### General


#### Server
#### Client

### Motivation and Context
When an exam receives a complaint, the dashboard shows this complaint as solved. 
![Screenshot 2022-03-23 at 17 21 45](https://user-images.githubusercontent.com/88991954/159779239-ef0312da-9be1-4261-965b-7ca63a71096b.png)

That is wrong because, without a response, the dashboard should show **0/1 (0%)**. Otherwise, the instructors will think that all complaints were solved. 

According to the presentation of the information, some minor corrections are applied. 

### Description
Now, every time a complaint is submitted, the counting is correct. 

- If there are 2 complaints without an answer, the dashboard will show 0/2 (0%). 
- If one of these is answered, the dashboard will show 1/2 (50%). 
- If all complaints were answered -> 2/2 (100%)   

**Just like the exercises dashboard works.** 

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming/Text/Modeling Exercise with Complaints enabled

1. Log in to Artemis
2. Navigate to Course Administration
3. Create an exam
4. Add a couple of exercises with enabled complaints (at least 2)
5. The student submits the exam
6. The instructor makes the assessment 
7. The student submits a complaint to each exercise
8. Verify that the 2 complaints are shown in the dashboard 0/2 (0%)
9. Answer one complaint and verify the dashboard shows Complaints: 1/2 (50%)
10. Answer the second complaint and verify the dashboard shows Complaints: 2/2 (100%)

Reviewers can use the number of exercises and complaints they prefer. They just need to verify that the counting is correct. 

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|


### Screenshots
